### PR TITLE
parser: Support the ODBC syntax of time/date/timestamp literal

### PR DESCRIPTION
**Pull request by shenli**
Fix the issue in tidb user google group.
https://groups.google.com/forum/#!topic/tidb-user/wB_U4NDw_90

### CODE REVIEW

**Comment by zz-jason** [comment]
@tiancaiamao PTAL

File: parser/parser_test.go [hunk 1]
```diff
@@ -618,6 +618,16 @@ func (s *testParserSuite) TestExpression(c *C) {
 		// for timestamp literal
 		{"select timestamp '1989-09-10 11:11:11'", true},
 		{"select timestamp 19890910", false},
+
+		// The ODBC syntax for time/date/timestamp literal.
+		// See: https://dev.mysql.com/doc/refman/5.7/en/date-and-time-literals.html
+		{"select {ts '1989-09-10 11:11:11'}", true},
+		{"select {d '1989-09-10'}", true},
+		{"select {t '00:00:00.111'}", true},
+		{"select {ts '1989-09-10 11:11:11'}", true},
```
Comment (tiancaiamao) [inline parser/parser_test.go]: The same with line 624?

File: parser/parser_test.go [hunk 2]
```diff
@@ -618,6 +618,16 @@ func (s *testParserSuite) TestExpression(c *C) {
...
+		// If the identifier is not in (t, d, ts), we just ignore it and consider the following expression as a string literal.
+		// This is the same behavior with MySQL.
+		{"select {ts '1989-09-10 11:11:11'}", true},
```
Comment (tiancaiamao) [inline parser/parser_test.go]: It should be `select {xx '1989-09-10 11:11:11'}`

**Comment by tiancaiamao** [comment]
LGTM
/run-all-tests

**Pull request merged**